From e0922a13eb5434939cc832d22806f57e83adf09c Mon Sep 17 00:00:00 2001
From: Jens Rottmann <JRottmann@LiPPERTembedded.de>
Date: Wed, 8 Jan 2014 16:26:51 +0100
Subject: Our PWM backlight has an enable pin but no CABC.

The SMARC 1.1 spec isn't really clear how LCD_BKLT_EN and LCD_BKLT_PWM are
related, they might be independent (as I had assumed in our previous Linux
3.10 BSP), but now I interpret _EN as enable signal and _PWM as brightness
control for one and the same backlight.

This patch adds LCD_BKLT_EN as enable GPIO to the existing backlight node.
Our LCD_BKLT_PWM is also muxed differently than SabreSD's. Default
brightness is set to 50%, that way the panel will be lit regardless of PWM
polarity.

We don't feature CABC (Content Adaptive Backlight Control), remove it.

diff --git a/arch/arm/boot/dts/lec-imx6.dtsi b/arch/arm/boot/dts/lec-imx6.dtsi
index c184978..be8cdb8 100644
--- a/arch/arm/boot/dts/lec-imx6.dtsi
+++ b/arch/arm/boot/dts/lec-imx6.dtsi
@@ -22,16 +22,6 @@
 		mxcfb3 = &mxcfb4;
 	};
 
-	hannstar_cabc {
-		compatible = "hannstar,cabc";
-		lvds0 {
-			gpios = <&gpio6 15 GPIO_ACTIVE_HIGH>;
-		};
-		lvds1 {
-			gpios = <&gpio6 16 GPIO_ACTIVE_HIGH>;
-		};
-	};
-
 	chosen {
 		stdout-path = &uart1;
 	};
@@ -271,7 +261,8 @@
 		compatible = "pwm-backlight";
 		pwms = <&pwm1 0 5000000>;
 		brightness-levels = <0 4 8 16 32 64 128 255>;
-		default-brightness-level = <7>;
+		default-brightness-level = <6>;
+		enable-gpios = <&gpio6 10 GPIO_ACTIVE_HIGH>; /* SMARC LCD_BKLT_EN */
 		status = "okay";
 	};
 
@@ -872,7 +863,8 @@
 
 		pinctrl_pwm1: pwm1grp {
 			fsl,pins = <
-				MX6QDL_PAD_SD1_DAT3__PWM1_OUT		0x1b0b1
+				MX6QDL_PAD_NANDF_RB0__GPIO6_IO10     0x80000000 /* SMARC LCD_BKLT_EN */
+				MX6QDL_PAD_GPIO_9__PWM1_OUT		0x1b0b1 /* SMARC LCD_BKLT_PWM */
 			>;
 		};
 
@@ -1038,7 +1030,7 @@
 	status = "okay";
 };
 
-&pwm1 {
+&pwm1 {		/* SMARC LCD_BKLT_PWM */
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_pwm1>;
 	status = "okay";
-- 
2.1.4

